### PR TITLE
Wrap call to scikit-learn's partial dependence method in a try/finally block

### PR DESCRIPTION
**Pull request by angela97lin**
From @dsherry's comment in #1150, wrap call to scikit-learn's partial dependence method in a try/finally block so that we still delete the set attributes even if call to scikit-learn's partial dependence method fails or errors out.

### CODE REVIEW

**Review by jeremyliweishih — APPROVED** [review]
👍

File: evalml/tests/model_understanding_tests/test_graphs.py [hunk 1]
```diff
@@ -689,6 +689,20 @@ def test_partial_dependence_problem_types(problem_type, X_y_binary, X_y_multi, X
         pipeline.feature_importances_
 
 
+@patch('evalml.model_understanding.graphs.sk_partial_dependence')
+def test_partial_dependence_error_still_deletes_attributes(mock_part_dep, X_y_binary, logistic_regression_binary_pipeline_class):
```
Comment (bchen1116) [inline evalml/tests/model_understanding_tests/test_graphs.py:693]: Nice test!

**Review by bchen1116 — APPROVED** [review]
LGTM!

File: evalml/model_understanding/graphs.py [hunk 2]
```diff
@@ -437,11 +437,12 @@ def partial_dependence(pipeline, X, feature, grid_resolution=100):
     elif isinstance(pipeline, evalml.pipelines.RegressionPipeline):
         pipeline._estimator_type = "regressor"
     pipeline.feature_importances_ = pipeline.feature_importance
-    avg_pred, values = sk_partial_dependence(pipeline, X=X, features=[feature], grid_resolution=grid_resolution)
-
-    # Delete scikit-learn attributes that were temporarily set
-    del pipeline._estimator_type
-    del pipeline.feature_importances_
+    try:
```
Comment (freddyaboulton) [inline evalml/model_understanding/graphs.py:440]: If we have to repeat this pattern in the future, it'd be cool to use a context manager!

**Review by freddyaboulton — APPROVED** [review]
@angela97lin I think this looks great!

**Pull request merged**